src/parsers/forecast_parser.jl

```diff
@@ -56,171 +56,55 @@ end
             end
         else
-            @warn "No forecast found for Loads"
-        end
-    end
-    return forecast
-end
-
-
-"""
-Args:
-    Dictionary of all the data files
-    Length of the forecast - Week()/Dates.Day()/Dates.Hour()
-    Forecast horizon in hours - Int64
-    Array of PowerSystems devices in the systems- Loads
-    Array of PowerSystems LoadZones
-
-Returns:
-    Returns an dictionary with Device name as key and PowerSystems Forecasts
-    dictionary as values
-"""
-function make_forecast_dict(time_series::Dict{String,Any},
-                            resolution::Dates.Period, horizon::Int,
-                            Devices::Array{ElectricLoad,1},
-                            LoadZones::Array{Device,1})
-    forecast = Dict{String,Any}()
-    for device in Devices
-        if haskey(time_series,"load")
-            for lz in LoadZones
-                if device.bus in lz.buses
-                    df = time_series["load"][[:DateTime,Symbol(lz.name)]]
-                    forecast[device.name] = make_device_forecast(device, df, resolution, horizon)
+            devices = _get_device(key,sys) #retrieve the device object
+            cn = isa(ts,DataFrames.DataFrame) ? names(ts) : TimeSeries.colnames(ts)
+            cn = [c for c in cn if c != :DateTime]
+        
+            if length(devices) > 0 
+                devices = unique(values(devices))
+                for d in devices
+                    for c in cn #if a TimeArray has multiple value columns, create mulitiple forecasts for different parameters in the same device
+                        timeseries = isa(ts,DataFrames.DataFrame) ? TimeSeries.TimeArray(ts.DateTime,ts[c]) : ts[c]
+                        push!(fc,Deterministic(d,string(c),timeseries))
+                    end
                 end
+            else
+                @warn("no $key entries for devices in sys")
             end
-        else
-            @warn "No forecast found for Loads"
         end
     end
-    return forecast
-end
+    return fc
+ end
 
 
-# - Parse Dict to Forecast Struct
+# - Assign Forecast to System Struct
 
 """
 Args:
-    A PowerSystems forecast dictionary
+    A System struct
+    A :Symbol=>Array{ <: Forecast,1} Pair denoting the forecast name and array of device forecasts
 Returns:
-    A PowerSystems forecast stuct array
+    A System struct with a modeified forecasts field
 """
-function make_forecast_array(dict)
-    Forecasts = Array{Forecast}(undef, 0)
-    for (device_key,device_dict) in dict
-                push!(Forecasts,Deterministic(device_dict["device"],device_dict["horizon"],
-                                device_dict["resolution"],device_dict["interval"],
-                                device_dict["initialtime"],
-                                device_dict["data"]
-                                ))
-    end
-    return Forecasts
+function add_forecast!(sys::System,fc::Pair{Symbol,Array{Forecast,1}})
```

Comment:
I have a preference for putting functions that modify a struct in the same file as that struct.  Thoughts?  Also, you have a typo in the docstring - "modeified".